tests/ert/unit_tests/forward_model_runner/test_fm_dispatch.py

```diff
@@ -405,3 +407,94 @@ def raises(*args, **kwargs):
 
     _report_all_messages(iter([message1, message2]), [reporter])
     reporter.report.assert_called_once_with(message1)
+
+
+@pytest.mark.integration_test
+async def test_fm_dispatch_sends_exited_event_with_terminated_msg_on_sigterm(
+    tmp_path, unused_tcp_port
+):
+    os.chdir(tmp_path)
+    with open("dummy_executable", "w", encoding="utf-8") as f:  # noqa: ASYNC230
+        f.write(
+            """#!/usr/bin/env python
+import time
+time.sleep(180)"""
+        )
+
+    executable = os.path.realpath("dummy_executable")
+    os.chmod("dummy_executable", stat.S_IRWXU | stat.S_IRWXO | stat.S_IRWXG)
+
+    fm_description = {
+        "global_environment": {},
+        "global_update_path": {},
+        "ens_id": "_id_",
+        "dispatch_url": f"tcp://localhost:{unused_tcp_port}",
+        "jobList": [
+            {
+                "name": "dummy_executable",
+                "executable": executable,
+                "target_file": None,
+                "error_file": None,
+                "start_file": None,
+                "stdout": "dummy.stdout",
+                "stderr": "dummy.stderr",
+                "stdin": None,
+                "argList": ["0"],
+                "environment": None,
+                "license_path": None,
+                "max_running_minutes": None,
+                "min_arg": 0,
+                "arg_types": [],
+                "max_arg": None,
+            }
+        ],
+        "run_id": "",
+        "ert_pid": "",
+    }
+
+    with open(FORWARD_MODEL_DESCRIPTION_FILE, "w", encoding="utf-8") as f:  # noqa: ASYNC230
+        f.write(json.dumps(fm_description))
+
+    # macOS doesn't provide /usr/bin/setsid, so we roll our own
+    with open("setsid", "w", encoding="utf-8") as f:  # noqa: ASYNC230
+        f.write(
+            dedent(
+                """\
+            #!/usr/bin/env python
+            import os
+            import sys
+            os.setsid()
+            os.execvp(sys.argv[1], sys.argv[1:])
+            """
+            )
+        )
+    os.chmod("setsid", 0o755)
+
+    async with MockZMQServer(unused_tcp_port) as zmq_server:
+        fm_dispatch_process = Popen(  # noqa: ASYNC220
+            [
+                os.getcwd() + "/setsid",
+                "fm_dispatch.py",
+                os.getcwd(),
+            ]
+        )
+        p = psutil.Process(fm_dispatch_process.pid)
+
+        async def wait_for_msg(msg_type):
+            while True:
+                await asyncio.sleep(0.5)
+                if any(
+                    msg_type in orjson.loads(msg).get("event_type", "")
+                    for msg in zmq_server.messages
+                ):
+                    return
+
+        # wait for fm running
+        await asyncio.wait_for(wait_for_msg("forward_model_step.running"), timeout=10)
+        p.terminate()
```

Comment:
How safe is it to test process terminate? Maybe @berland can double check?

Reply:
should be the same as `os.kill()` (with sigterm I suppose). If it works as intended, I would be happy :)